lib/backend/key.go

```diff
@@ -19,81 +19,138 @@ package backend
 import (
 	"bytes"
 	"fmt"
+	"slices"
 	"strings"
 )
 
 // Key is the unique identifier for an [Item].
-type Key []byte
+type Key struct {
+	components []string
+	s          string
+	exactKey   bool
+	noEnd      bool
+}
 
 // Separator is used as a separator between key parts
 const Separator = '/'
+const SeparatorString = string(Separator)
 
 // NewKey joins parts into path separated by Separator,
 // makes sure path always starts with Separator ("/")
-func NewKey(parts ...string) Key {
-	return internalKey("", parts...)
+func NewKey(components ...string) Key {
+	k := internalKey("", components...)
+	k.exactKey = k.s == SeparatorString || (len(k.s) > 0 && k.s[len(k.s)-1] == Separator)
```

Comment:
```suggestion
	k.exactKey = len(k.s) > 0 && k.s[len(k.s)-1] == Separator
```

Isn't `internalKey` always returning a nonempty key, anyway?